packages/grafana-runtime/src/services/templateSrv.ts

```diff
@@ -0,0 +1,13 @@
+import { VariableModel } from '@grafana/data';
+
+export interface TemplateSrv {
+  getVariables(): VariableModel[];
```

Comment:
This is great work @ryantxu , the bigger philosophical question here is how to expose Redux variables to plugin authors, I think. `getVariables` will use `getState` which will call store.getState directly and I'm not sure that's the way we want to expose data from our Redux store. 

For instance, if we  would use `getVariables` in a reducer in Grafana, things would blow up. Maybe we should exposa a subscription of VariableModel instead, then it's more like how you would go about and subscribe to the store? But I've never done this before, so I'm uncertain what is the best way.

Reply:
> the bigger philosophical question

HA... yes we have dragged our feet for a very long time finding something appropriate.  Given that we do not expose (nor do I think we should expose) the internal redux store, I think this is safe.

A subscription would be interesting -- but lets try that later ;) most of the time this is used to add a few options to a dropdown in query editors